src/main/java/ch/njol/skript/expressions/ExprArrowPierceLevel.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * This file is part of Skript.
+ *
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.entity.AbstractArrow;
+import org.bukkit.entity.Arrow;
+import org.bukkit.entity.Projectile;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Arrow Pierce Level")
+@Description("An arrow's pierce level.")
+@Examples({"on shoot:",
+	"\tevent-projectile is an arrow",
+	"\tset arrow pierce level of event-projectile to 5"})
+@Since("INSERT VERSION")
+public class ExprArrowPierceLevel extends SimplePropertyExpression<Projectile, Number> {
+	
+	static {
+		if(Skript.classExists("org.bukkit.entity.AbstractArrow") || Skript.methodExists(Arrow.class, "getPierceLevel")) register(ExprArrowPierceLevel.class, Number.class, "[the] arrow pierce level", "projectiles");
+	}
+	
+	boolean abstractArrowExists = Skript.classExists("org.bukkit.entity.AbstractArrow");
+	
+	@Nullable
+	@Override
+	public Number convert(Projectile arrow) {
+		if (abstractArrowExists)
+			return arrow instanceof AbstractArrow ? ((AbstractArrow) arrow).getPierceLevel() : null;
+		return arrow instanceof Arrow ? ((Arrow) arrow).getPierceLevel() : null;
+	}
+	
+	@Nullable
+	@Override
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		return (mode == ChangeMode.SET || mode == ChangeMode.RESET) ? CollectionUtils.array(Number.class) : null;
```

Comment:
Should also support add and remove changers considering you are able to use those this way:
```
increase shot arrow's pierce level by 1
```
Forgot to mention this on your other PR for knockback strength